### ISSUE TITLE
[import/no-duplicates] `{"prefer-inline": true}` does not work

### ISSUE THREAD

**Issue by johnvonneumann7** [import-js/eslint-plugin-import#2715]
version `2.27.5`

https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md#inline-type-imports
The documentation states that adding `{"prefer-inline": true}` as shown below will cause errors and autofixes if there are duplicate imports. However, this was not possible in my environment.

```json
{
  "import/no-duplicates": ["error", {"prefer-inline": true}],
}
```
It should look like the following, but there is no error or anything.
```typescript
import NextErrorComponent from 'next/error';
import { type ErrorProps } from 'next/error';
// ↓
import NextErrorComponent, { type ErrorProps } from 'next/error';
```
![image](https://user-images.githubusercontent.com/78252058/218293767-a5655005-c670-4751-a430-ce0e780e7ec1.png)

Is this currently being corrected?

**Comment by ljharb** [comment]
I agree this should be autofixed - as well as warned on. cc @snewcomer

**Comment by snewcomer** [comment]
Ok perfect.  Been working on and throwing away solutions for a few weeks now here and there.  Will try to put up my progress today.

**Comment by guillaumebrunerie** [comment]
Related issue here, when running ESLint from the terminal I get the error

	Configuration for rule "import/no-duplicates" is invalid:
	Value {"prefer-inline":true} should NOT have additional properties.

even though I used the exact same options as in the docs.

**Comment by merrywhether** [comment]
The issue lies in [the creation of the various import maps](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-duplicates.js#L314-L332) in the rule. In the case of the following:

```ts
import { type Foo } from './bar';
import { foo } from './bar';
```
Line 1 will be sorted into `namedTypesImported` and Line 2 will be imported into `imported`, and then when the `checkImports` function is run later [neither of each of the maps' nodes will have length > 1](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-duplicates.js#L12). Currently the rule's example only works because both lines have type imports in them.

If I [wrap this block](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-duplicates.js#L327-L329) in a `if (!prefersInline)` check, the above correctly collapses into

```ts
import { foo , type Foo } from './bar';
```

I need to dig more though as I'm not yet sure which is the correct description of the problem:
1. It's not correct to ever sort the inline imports into the `*TypesImported` maps because those are for managing `import type ...` type-style import statements which can't be directly compared with `import ...` value-style import statements. In this case, the wrapped block above is not necessary at all?
2. Imports with inline type imports should get added to multiple maps. Then we compare types against all types and values against all values, but I'm not sure what happens in a multi-way tie (I haven't grokked the full rule enough yet).

**Comment by snewcomer** [comment]
@merrywhether Could you try and symlink this PR?  I still have to run it against a few repos to see if it works but it would be good to get some early signal!


https://github.com/import-js/eslint-plugin-import/pull/2716

**Comment by snewcomer** [comment]
@guillaumebrunerie 

> Related issue here, when running ESLint from the terminal I get the error

Have you updated the eslint-plugin-import to a version that it is supported.  We had an issue where the docs were published before the feature was released.  In any case, I would avoid upgrading anyways.  I made a mistake with the original PR where I still had the typescript-eslint repo linked where I did the bundled work so got a false positive that the amalgamation of PRs worked :(.  `no-duplicates` It is pretty broken right now.

**Comment by ljharb** [comment]
> We had an issue where the docs were published before the feature was released.

That's not entirely accurate; docs are always landed on main prior to being released - users need to check the tagged docs (on every repo on github, not just this one)

**Comment by guillaumebrunerie** [comment]
Oh, it can very well be that I did not upgrade, I didn't know it was a new feature. But in that case the error message is very confusing, I think it should say something like `Unknown rule "import/no-duplicates".` or similar.

**Comment by ljharb** [comment]
@guillaumebrunerie you'd need to ask eslint about that; we have no control over that error message. In this case, the rule was known, but that schema option was not.

**Comment by niklasnatter** [comment]
I think this is similar to https://github.com/import-js/eslint-plugin-import/issues/2675

**Comment by kevinwolfcr** [comment]
Having this issue too. It only detects an error when an existing import already has types, like this one:

```ts
import { AValue, type AType } from './mama-mia'
import type { BType } from './mama-mia'
```

Gets fixed into this:

```ts
import { type AType, AValue, type BType } from "./mama-mia"
```

However, if first import doesn't have a type import already, like this:

```ts
import { AValue } from './mama-mia'
import type { BType } from './mama-mia'
```

It doesn't work on that case.

**Comment by alecmev** [comment]
Fixed by https://github.com/import-js/eslint-plugin-import/pull/2835?

**Comment by dylang** [comment]
related? https://github.com/import-js/eslint-plugin-import/issues/2792

**Comment by juunzzi** [comment]
help me please..
